### DISCUSSION TITLE
[Question] How to generate horovod timeline avoid preprocessing ?

### DISCUSSION THREAD

**Discussion by Nov11** [horovod/horovod#3893] [Q&A]
I have a training task that does a lot of preprocessing before starting actual training.
I want to avoid recording the preprocessing part, but only later part.
How can I achieve this?

BTW, if the long running task is start with --timeline-filename xxx.json and stop later using Ctrl+c, it seems that the timeline file is always broken.
Is it possible to get timeline from a long running task before it naturally exist?

**Comment by maxhgerlach** [comment]
There is `hvd.start_timeline()` and `hvd.stop_timeline()`. https://github.com/horovod/horovod/blob/2ef8ff92f541892108bfaa3b9879986203bc297a/horovod/common/basics.py#L159-L184

Would that work for you?
